lib/config.js

```diff
@@ -127,6 +141,22 @@ var normalizeConfig = function (config, configFilePath) {
   // normalize urlRoot
   config.urlRoot = normalizeUrlRoot(config.urlRoot)
 
+  // normalize and default upstream proxy settings if given
+  if (config.upstreamProxy) {
+    var proxy = config.upstreamProxy
+    proxy.path = proxy.path ? normalizeProxyPath(proxy.path) : '/'
+    proxy.hostname = proxy.hostname || 'localhost'
```

Comment:
this should be a check for `_.isUndefined(proxy.hostname)` as this precludes someone from setting the `hostname` to `''`